ODIN_II/SRC/ast_util.cpp

```diff
@@ -1045,9 +1047,28 @@ ast_node_t *resolve_node(STRING_CACHE_LIST *local_string_cache_list, ast_node_t
 							error_message(NETLIST_ERROR, node->line_number, node->file_number, "Parameter %s is not a constant expression\n", node->types.identifier);
 						}
 						node = newNode;
-						return node;
 					}
+					else
+					{
+						break;
+					}
+				}
+				else
+				{
+					break;
+				}
+			}
+			// fallthrough
+
+			case NUMBERS:
+			{
+				if (top_case && assignment_size > 0) 
```

Comment:
I would simply check that (max_size == &my_max) since the root case is responsible for setting the initial pointer